tap_appstore/discover.py

```diff
@@ -16,6 +18,10 @@ def do_discover(client: Api) -> Catalog:
     return catalog
 
 
+@retry(
+    stop=stop_after_attempt(5),
```

Comment:
Retry on specific errors. E.g. `'API call failed - List of users is empty'` is useless to retry.

Reply:
I've set `retry_if_exception_type(APIError)` and added a test to make sure the 'API call failed - List of users is empty' will not trigger another retry.